data_source.go

```diff
@@ -303,38 +299,42 @@ func (ds *pgDataSource) buildConnInfo(conninfo map[string]string) string {
 		ds.cfg.Database, ds.cfg.User, ds.cfg.Sslmode, ds.cfg.BinaryParameters, ds.cfg.Password)
 }
 
-func (ds *pgDataSource) getPgCurrentWalLsn(role string) (string, error) {
-	if role == "replica" {
-		// query select * from pg_stat_wal_receiver;  to get conninfo
-		//create a new db connection to upstream (primary) with conninfo and return pg_current_wal_lsn
-		conninfo, err := ds.getUpstreamConnInfo()
-		if err != nil {
-			return "", err
-		}
-		upstreamConnInfo := ds.buildConnInfo(parseConnInfo(conninfo))
-		upstreamDb, err := sqlConnect(upstreamConnInfo)
-		if err != nil {
-			return "", err
-		}
-		defer upstreamDb.Close()
-		var pgCurrentWalLsn string
-		err = upstreamDb.Get(&pgCurrentWalLsn, "select pg_current_wal_lsn()")
-		if err != nil {
-			return "", err
+func (ds *pgDataSource) getPgCurrentWalLsn(maxHop int64, db *sqlx.DB) (string, error) {
+	var isReplica bool
+	err := db.Get(&isReplica, "select pg_catalog.pg_is_in_recovery()")
+	defer db.Close()
```

Comment:
Do we need to close the DB pool here?

Reply:
Oh, I see.. you're passing the upstream pool into the function so it will need to be closed or we'll be in trouble. That's good, but we probably don't want to close the `ds.db` pool.

Reply:
ah good point, we don't want to close `ds.db` pool. I'll remove that line.

Reply:
as discussed earlier:
https://github.com/film42/pgreba/blob/f39c543f226c9d8aa421570c8ffa6268db1272bd/data_source.go#L320-L323